### ISSUE TITLE
Uncaught error from thread [application-scheduler-1]: GC overhead limit exceeded, shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[application]

### ISSUE THREAD

**Issue by PatrickF-mbse** [gorenje/sysmlv2-jupyter-docker#15]
Hi,

I tried to run a publish command and got the error below.   I found this post about it (https://stackoverflow.com/questions/1393486/error-java-lang-outofmemoryerror-gc-overhead-limit-exceeded) advising to   "Increase the heap size, for example -Xmx1g"  but i am not sure where I should run this. inside the docker Sysml api server? on windows? on jupiterlab?

thanks if someone has an idea about the fix.  

**Uncaught error from thread [application-scheduler-1]: GC overhead limit exceeded, shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[application]**

**Note: in Jupiterlab, i am getting:**
API base path:   http://sysmlapiserver:9000/
Processing........................................
Posting Commit (7490 elements)...
Error: 400 Bad Request
Saved to Project IX5UASMM_V2 Thu Jul 27 00:38:41 UTC 2023 (30e16fab-ffbb-46d4-9424-e53b743e5ed1)

**Log from the docker container:**
2023-07-27 00:39:38 Uncaught error from thread [application-scheduler-1]: GC overhead limit exceeded, shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[application]
2023-07-27 00:39:38 java.lang.OutOfMemoryError: GC overhead limit exceeded
2023-07-27 00:39:38 [ERROR] [07/27/2023 00:39:38.659] [sbt-web-scheduler-1] [akka.actor.ActorSystemImpl(sbt-web)] exception on LARS’ timer thread
2023-07-27 00:39:38 java.lang.OutOfMemoryError: GC overhead limit exceeded
2023-07-27 00:39:38 
2023-07-27 00:39:38 [INFO] [07/27/2023 00:39:38.662] [sbt-web-scheduler-1] [akka.actor.ActorSystemImpl(sbt-web)] starting new LARS thread
2023-07-27 00:39:38 [error] a.a.ActorSystemImpl - exception on LARS’ timer thread
2023-07-27 00:39:38 java.lang.OutOfMemoryError: GC overhead limit exceeded
2023-07-27 00:39:38 Uncaught error from thread [[error] a.a.ActorSystemImpl - Uncaught error from thread [application-scheduler-1]: GC overhead limit exceeded, shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[application]
2023-07-27 00:39:38 java.lang.OutOfMemoryError: GC overhead limit exceeded
2023-07-27 00:39:38 sbt-web-scheduler-1]: GC overhead limit exceeded, shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[sbt-web]
2023-07-27 00:39:38 java.lang.OutOfMemoryError: GC overhead limit exceeded
2023-07-27 00:39:38 [ERROR] [SECURITY][07/27/2023 00:39:38.673] [sbt-web-scheduler-1] [akka.actor.ActorSystemImpl(sbt-web)] Uncaught error from thread [sbt-web-scheduler-1]: GC overhead limit exceeded, shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[sbt-web]
2023-07-27 00:39:38 java.lang.OutOfMemoryError: GC overhead limit exceeded
2023-07-27 00:39:38 
2023-07-27 00:39:38 [info] p.c.s.AkkaHttpServer - Stopping server...
2023-07-27 00:39:44 [info] Loading settings for project sysml-v2-api-services-2023-02-build from scaffold.sbt,plugins.sbt ...
2023-07-27 00:39:45 [info] Loading project definition from /home/sysml/SysML-v2-API-Services-2023-02/project
2023-07-27 00:39:47 [info] Loading settings for project root from build.sbt ...
2023-07-27 00:39:47 [info] Set current project to SysML-v2-API-Services (in build file:/home/sysml/SysML-v2-API-Services-2023-02/)
2023-07-27 00:39:49 
2023-07-27 00:39:50 --- (Running the application, auto-reloading is enabled) ---
2023-07-27 00:39:50 
2023-07-27 00:39:51 [info] p.c.s.AkkaHttpServer - Listening for HTTP on /0.0.0.0:9000
2023-07-27 00:39:51 
2023-07-27 00:39:51 (Server started, use Enter to stop and go back to the console...)
2023-07-27 00:39:51

**Comment by gorenje** [comment]
```
2023-07-27 00:39:47 [info] Set current project to SysML-v2-API-Services (in build file:/home/sysml/SysML-v2-API-Services-2023-02/)
```

that appears to be the [API server](https://github.com/Systems-Modeling/SysML-v2-API-Services) which does utilise the JVM. The [Dockerfile.api](https://github.com/gorenje/sysmlv2-jupyter-docker/blob/main/Dockerfile.api) builds the api server but in the end it just calls [`sbt run`](https://github.com/gorenje/sysmlv2-jupyter-docker/blob/main/Dockerfile.api#L68) whatever that does or however it is configured is beyond my understanding.

These are the [instructions](https://github.com/Systems-Modeling/SysML-v2-API-Services#setup-sbt) I followed setting up the dockerfile. I make [some changes](https://github.com/gorenje/sysmlv2-jupyter-docker/blob/main/Dockerfile.api#L55-L65) to the underlying setup but none that would affect the JVM. How one gets at the JVM options isn't clear to me.

**Comment by PatrickF-mbse** [comment]
@gorenje     Hi,  thank you for the follow up.  I tried to play with the RAM allocated to the docker VM but it did not help. I did a few test and everything work fine with model up to 3700 elements then anything higher cause this error.   I will  limit the model size for now as I also have no idea. I will try on a better computer later to see if it s help or try to publish from eclipse to the docker api server.    Thank you anyway.
